### PR TITLE
Add support for different search inputs when searching for patient

### PR DESCRIPTION
**Pull request by vinaysshenoy**
Pivotal Story Link: https://www.pivotaltracker.com/story/show/167343626

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/addidtopatient/searchresults/AddIdToPatientSearchResultsScreen.kt [hunk 1]
```diff
@@ -110,7 +111,9 @@ class AddIdToPatientSearchResultsScreen(context: Context, attrs: AttributeSet) :
 
   private fun screenCreates(): Observable<UiEvent> {
     val screenKey = screenRouter.key<AddIdToPatientSearchResultsScreenKey>(this)
-    searchResultsView.downstreamUiEvents.onNext(SearchResultPatientName(screenKey.fullName))
+    searchResultsView
+        .downstreamUiEvents
+        .onNext(SearchPatientCriteria(SearchPatientBy.Name(screenKey.fullName)))
```
Comment (ragunathjawahar) [inline app/src/main/java/org/simple/clinic/addidtopatient/searchresults/AddIdToPatientSearchResultsScreen.kt]: Nice!

File: app/src/main/java/org/simple/clinic/search/results/PatientSearchResultsScreen.kt [hunk 2]
```diff
@@ -74,7 +75,14 @@ class PatientSearchResultsScreen(context: Context, attrs: AttributeSet) : Relati
...
         .upstreamUiEvents
         .ofType<RegisterNewPatient>()
-        .map { PatientSearchResultRegisterNewPatient(it.patientName) }
+        .map { PatientSearchResultRegisterNewPatient(extractPatientName(it.searchBy)) }
+  }
+
+  private fun extractPatientName(searchBy: SearchPatientBy): String {
+    return when (searchBy) {
+      is SearchPatientBy.Name -> searchBy.searchText
+      is SearchPatientBy.PhoneNumber -> TODO("not yet implemented")
```
Comment (ragunathjawahar) [inline app/src/main/java/org/simple/clinic/search/results/PatientSearchResultsScreen.kt]: Is there a possibility for this branch to get executed at the moment?

File: app/src/main/java/org/simple/clinic/searchresultsview/SearchPatientBy.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package org.simple.clinic.searchresultsview
+
+sealed class SearchPatientBy {
```
Comment (ragunathjawahar) [inline app/src/main/java/org/simple/clinic/searchresultsview/SearchPatientBy.kt]: Nice!

File: app/src/main/java/org/simple/clinic/search/results/PatientSearchResultsScreen.kt (hunk 2 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/search/results/PatientSearchResultsScreen.kt]: Nope, the screen this view is embedded in will only invoke this with the name criteria at the moment.

**Pull request merged**